main.py

```diff
@@ -23,102 +27,38 @@
-    time.sleep(1)
-    outeriframe.click()
-    time.sleep(2)
-    allIframesLen = driver.find_elements_by_tag_name('iframe')
-    time.sleep(1)
-    audioBtnFound = False
-    audioBtnIndex = -1
-    for index in range(len(allIframesLen)):
-        driver.switch_to.default_content()
-        iframe = driver.find_elements_by_tag_name('iframe')[index]
-        driver.switch_to.frame(iframe)
-        driver.implicitly_wait(2)
-        try:
-            audioBtn = driver.find_element_by_id(RECAPTCHA_AUDIO_BUTTON) or driver.find_element_by_id(RECAPTCHA_ANCHOR)
-            audioBtn.click()
-            audioBtnFound = True
-            audioBtnIndex = index
-            break
-        except Exception as e:
-            pass
-    if audioBtnFound:
-        try:
-            while True:
-                href = driver.find_element_by_id(AUDIO_SOURCE).get_attribute('src')
-                response = requests.get(href, stream=True)
-                saveFile(response, CAPTCHA_MP3_FILENAME)
-                response = audioToText(CAPTCHA_MP3_FILENAME)
-                print(response)
-                driver.switch_to.default_content()
-                iframe = driver.find_elements_by_tag_name('iframe')[audioBtnIndex]
-                driver.switch_to.frame(iframe)
-                inputbtn = driver.find_element_by_id(AUDIO_RESPONSE)
-                inputbtn.send_keys(response)
-                inputbtn.send_keys(Keys.ENTER)
-                time.sleep(2)
-                errorMsg = driver.find_elements_by_class_name(AUDIO_ERROR_MESSAGE)[0]
-                if errorMsg.text == "" or errorMsg.value_of_css_property('display') == 'none':
-                    print("reCaptcha defeated!")
-                    break
-        except Exception as e:
-            print(e)
-            print('Oops, something happened. Check above this message for errors or check the chrome window to see if captcha locked you out...')
-    else:
-        print('Button not found. This should not happen.')
-
-    time.sleep(2)
-    driver.switch_to.default_content()
-
 def start_driver(random_city):
-    driver = webdriver.Chrome(ChromeDriverManager().install())
+    options = Options()
+    if (args.debug == DEBUG_DISABLED):
+        options.add_argument(f"user-agent={USER_AGENT}")
+        options.add_argument('disable-blink-features=AutomationControlled')
+        options.headless = True
+        driver = webdriver.Chrome(options=options)
+        driver.set_window_size(1440, 900)
+    elif (args.debug == DEBUG_ENABLED):
+        driver = webdriver.Chrome()
     driver.get(CITIES_TO_URLS[random_city])
     driver.implicitly_wait(10)
-    time.sleep(2)
+    WebDriverWait(driver, 10).until(expected_conditions.presence_of_element_located((By.XPATH, CREATE_AN_ACCOUNT_BUTTON)))
     driver.find_element_by_xpath(APPLY_NOW_BUTTON_1).click()
     driver.find_element_by_xpath(APPLY_NOW_BUTTON_2).click()
     driver.find_element_by_xpath(CREATE_AN_ACCOUNT_BUTTON).click()
```

Comment:
When I try running this, line 60 runs just fine (with `--debug` enabled, the window opens and the page loads), but something between lines 60 and 65 fails and causes the code to crash. I surrounded it with a `try...except`, which gave me the very helpful error message `Message: `. I'm running on Mac - are y'all using PC? Maybe that's part of the problem?

Reply:
Did more tinkering. Line `61` was the problem, I removed the `WebDriverWait` call and was able to fill out and submit an application. I think it could do with some verbosity while it's waiting for the email to come through so the user isn't left wondering what's happening, but otherwise this works great. Incredible job!